apps/src/lib/config/ethereum_bridge/params.rs

```diff
@@ -0,0 +1,115 @@
+//! Blockchain-level parameters for the configuration of the Ethereum bridge.
+use std::num::NonZeroU64;
+
+use serde::{Deserialize, Serialize};
+
+/// Represents a configuration value for an Ethereum address.
+///
+/// For instance:
+/// `0x6B175474E89094C44Da98b954EedeAC495271d0F`
+#[derive(Clone, Eq, PartialEq, Debug, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct Address(String);
+
+/// Represents a configuration value for the minimum number of
+/// confirmations an Ethereum event must reach before it can be acted on.
+#[derive(Clone, Copy, Eq, PartialEq, Debug, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct MinimumConfirmations(NonZeroU64);
+
+impl Default for MinimumConfirmations {
+    fn default() -> Self {
+        // SAFETY: The only way the API contract of `NonZeroU64` can be violated
+        // is if we construct values of this type using 0 as argument.
+        Self(unsafe { NonZeroU64::new_unchecked(100) })
+    }
+}
+
+/// Represents chain parameters for the Ethereum bridge.
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Config {
```

Comment:
```suggestion
pub struct GenesisConfig {
```

Reply:
Confused me at first because I though this was something that could be configured when starting a node. Reading the later code, I realized that wasn't true.

Reply:
Have renamed to `GenesisConfig`